### ISSUE TITLE
Remove Jupyter Adapter

### ISSUE THREAD

**Issue by jmcphers** [posit-dev/positron#4579]
Positron currently ships with a Jupyter Adapter, which bridges the Jupyter protocol and Positron's own protocol for language runtime messages. This module requires us to have native bindings to ZeroMQ, via its dependency on [zeromq.js](https://github.com/zeromq/zeromq.js). This dependency has been very bad for us.

- it requires a lot of fussy work with `electron-rebuild` and some other sensitive tools in order to produce a zeromq library that can run in the same process as node; it is consequently responsible for many build problems
- it dramatically slows down builds since the native bindings must be compiled from source
- it requires a [patched version of zeromq.js](https://github.com/jmcphers/zeromq.js) since we have (repeatedly) had to patch the upstream project to work around problems
- it doesn't work on older Linux versions, such as Ubuntu 20 and Debian 11, due to glibc compatibility issues (https://github.com/posit-dev/positron/issues/3854 and several other issues)
- it is the reason we have issues with libsodium (#1411) and friends. 

Here is where we currently inject zeromq.js:

https://github.com/posit-dev/positron/blob/888bb65db7d070c21828e8bdb884001adf8dcb07/extensions/jupyter-adapter/scripts/postinstall.ts#L30-L40

The ZeroMQ dependency can be moved elsewhere (to a separate process) to avoid these problems; we can use a less needy protocol (such as websockets) for the last mile to the extension host.

**Comment by jmcphers** [comment]
In recent builds, the Jupyter Adapter is optional; once the new supervisor is sufficiently stable, we can remove the Jupyter Adapter entirely.

**Comment by Trevor-Reid** [comment]
@jmcphers 
In Workbench `2024.12.1+549.pro5` dailies, we updated to `Positron: 2025.01.1 build 7`. Now in the browser's developer console, we see the following errors:

```
Activating extension 'positron.jupyter-adapter' failed: No native build was found for platform=linux arch=x64 runtime=electron abi=115 uv=1 libc=glibc node=20.18.1 webpack=true
```

<img width="980" alt="Image" src="https://github.com/user-attachments/assets/20138779-0a87-4f99-9984-1d9ada79d2bf" />

**Comment by jmcphers** [comment]
I think you're seeing #5918 which is resolved in newer builds.

**Comment by Trevor-Reid** [comment]
> I think you're seeing [#5918](https://github.com/posit-dev/positron/pull/5918) which is resolved in newer builds.

I'm assuming this is not part of the `2025.01.1` branch builds that Workbench is taking in as part of the `Kousa Dogwood` patch release. Does https://github.com/posit-dev/positron/pull/5918 need to be ported to that release branch (if it is not there already) ?

**Comment by jmcphers** [comment]
That's right, #5918 is not on the 2025.01 branch. The warning is harmless, though if it's causing any trouble LMK and I'll backport the change.

**Comment by Trevor-Reid** [comment]
> That's right, [#5918](https://github.com/posit-dev/positron/pull/5918) is not on the 2025.01 branch. The warning is harmless, though if it's causing any trouble LMK and I'll backport the change.

It is not a priority to back port the fix at this time unless we find an issue with this as a root cause (so far no issue has been seen so far)

**Comment by juliasilge** [comment]
I suspect that #6077 was some weird problem with the Jupyter Adapter, so definitely worth getting this out in a timely manner.

> Activating extension 'positron.jupyter-adapter' failed: No native build was found for platform=win32 arch=x64 runtime=electron abi=128 uv=1 libc=glibc node=20.18.1 electron=32.2.6 webpack=true
loaded from: c:\Program Files\Positron\resources\app\extensions\jupyter-adapter
